x-pack/plugins/security_solution/public/common/components/navigation/solution_grouped_nav/solution_grouped_nav.tsx

```diff
@@ -163,5 +149,84 @@ export const SolutionGroupedNavComponent: React.FC<SolutionGroupedNavProps> = ({
     </>
   );
 };
-
 export const SolutionGroupedNav = React.memo(SolutionGroupedNavComponent);
+
+const SolutionNavItems: React.FC<SolutionNavItemsProps> = ({
+  items,
+  selectedId,
+  activePanelNavId,
+  isMobileSize,
+  navItemsById,
+  onOpenPanelNav,
+}) => (
+  <>
+    {items.map((item) => (
+      <SolutionNavItem
+        key={item.id}
+        item={item}
+        isSelected={selectedId === item.id}
+        isActive={activePanelNavId === item.id}
+        hasPanelNav={!isMobileSize && item.id in navItemsById}
+        onOpenPanelNav={onOpenPanelNav}
+      />
+    ))}
+  </>
+);
+
+const SolutionNavItemComponent: React.FC<SolutionNavItemProps> = ({
+  item,
+  isSelected,
+  isActive,
+  hasPanelNav,
+  onOpenPanelNav,
+}) => {
+  if (isCustomItem(item)) {
+    return <Fragment key={item.id}>{item.render(isSelected)}</Fragment>;
+  }
+  const { id, href, label, onClick } = item;
+
+  const itemClassNames = classNames('solutionGroupedNavItem', {
+    'solutionGroupedNavItem--isActive': isActive,
+    'solutionGroupedNavItem--isPrimary': isSelected,
+  });
+  const buttonClassNames = classNames('solutionGroupedNavItemButton');
+
+  const onButtonClick: React.MouseEventHandler = (ev) => {
+    ev.preventDefault();
+    ev.stopPropagation();
+    onOpenPanelNav(id);
+  };
+
+  return (
+    // eslint-disable-next-line @elastic/eui/href-or-on-click
+    <EuiLink
```

Comment:
LinkAnchor component ++

Reply:
good point, the thing is that I tried to use only generic components in the `solution_grouped_nav`. Just in case we want to make it available to other solutions later on, in the shared-ux plugin.